src/scripts/sync_teams/sync_teams.py

```diff
@@ -216,11 +246,13 @@ def main():
     assets: List[Asset] = list_assets(jit_token)
 
     teams_to_delete = process_teams(jit_token, organization, assets)
-
+    existing_teams: List[TeamAttributes] = get_existing_teams(jit_token)
```

Comment:
seems like the script is calling get_existing_teams 3 different times - is it intentional (expecting different response each time) or can the returned teams be reused? 
Otherwise, maybe the process_teams response can be extended and return existing_teams?

Reply:
It does for the second call since it's after we update the teams, HOWEVER the call in the handling of assets is indeed redundant, removed it.

Reply:
sound good.
I still believe the process_teams can return the existing teams without the need to refetch them (by returning the created teams when calling create_teams)